person.rb

```diff
@@ -0,0 +1,23 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age.to_i
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id, :parent_permission
```

Comment:
[OPTIONAL] I suggest you define your attribute accessor and reader before your constructor😉. even though it doesn't change anything, it is better since it is popularly known to be placed there and it is one of the first things one should see before going forward in your code and this will enhance code maintainability and readability😉.

Reply:
Thanks for your suggestion. 🙏